sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/CountingInput.java

```diff
@@ -157,7 +172,22 @@ public UnboundedCountingInput withTimestampFn(SerializableFunction<Long, Instant
     public UnboundedCountingInput withMaxNumRecords(long maxRecords) {
       checkArgument(
           maxRecords > 0, "MaxRecords must be a positive (nonzero) value. Got %s", maxRecords);
-      return new UnboundedCountingInput(timestampFn, Optional.of(maxRecords), maxReadTime);
+      return new UnboundedCountingInput(
+          timestampFn, elementsPerPeriod, period, Optional.of(maxRecords), maxReadTime);
+    }
+
+    /**
+     * Returns an {@link UnboundedCountingInput} like this one, but with output production limited
+     * to an aggregate rate of no more than the number of elements per the period length.
+     *
+     * <p>Note that this period is taken in aggregate across all instances of the
+     * {@link PTransform}, which may cause elements to be produced in bunches.
```

Comment:
We would prefer to use the point at which the pipeline starts to decide the actual starting point, but we don't have access to that value. We shouldn't use the time at which the source is created because there is an unknown amount of time between when the source is first created and when it is run.